element_array_ephys/readers/openephys.py

```diff
@@ -112,15 +137,23 @@ def __init__(self, processor, probe_index=0):
         self.processor_id = int(processor['@NodeId'])
         
         if processor['@pluginName'] == 'Neuropix-3a' or 'NP_PROBE' not in processor['EDITOR']:
-            self.probe_info = processor['EDITOR']['PROBE']
+            self.probe_info = processor['EDITOR']['PROBE'][probe_index]
             self.probe_SN = self.probe_info['@probe_serial_number']
             self.probe_model = {
                 "Neuropix-PXI": "neuropixels 1.0 - 3B",
                 "Neuropix-3a": "neuropixels 1.0 - 3A"}[processor['@pluginName']]
+            self._channels_connected = {int(re.search(r'\d+$', k).group()): int(v)
+                                        for k, v in self.probe_info.pop('CHANNELSTATUS').items()}
```

Comment:
```suggestion
                                        for k, v in self.probe_info['CHANNELSTATUS'].items()}
```

Reply:
`.pop()` is used here intentionally, so `self.probe_info` no longer contain `CHANNELSTATUS` (avoid redundant/non-relevant info in the returning object)